File: csharp-code/ExceptionHandling/UserDefinedException.cs
using System;
 
namespace Hello
{
    /*
        ExceptionHandling - user Defined Exceptions
        
        output
        
        Age should be greater than 18
        Exception of type 'Hello.InvalidAgeException' was thrown.
    */
    
    public class InvalidAgeException:Exception{
        
        public InvalidAgeException(String message):base(message)
        {
        }
    }
    public class Hello
    {
		public static void Main (string[] args)
		{
		    
		    try
		    {
		        int age=17;
    		    if(age<18)
    		        throw new InvalidAgeException("Age should be greater than 18");
    		    
		    }
		    catch(InvalidAgeException e)
		    {
		        Console.WriteLine(e.Message);
		    }
		}
	}
}

File: Java-notes/ExceptionHandling/UserDefinedException.java
/* 
 User Defined Exception 
 
  *** output *** 
  MyException: performing Integer Division
*/

class MyException extends Exception{
    public MyException(String message)
    {
        super(message);
    }
}

public class UserDefinedException{
    public static void main(String[] args)
    {
        int a=6,b=3;
        try{
            if(a%b==0)
                throw new MyException("performing Integer Division");   
        }
        catch(MyException e)
        {
            System.out.println(e);
        }
    }
}